sklearn/utils/extmath.py

```diff
@@ -885,51 +883,6 @@ def svd_flip(u, v, u_based_decision=True):
     return u, v
 
 
-def log_logistic(X, out=None):
-    """Compute the log of the logistic function, ``log(1 / (1 + e ** -x))``.
-
-    This implementation is numerically stable because it splits positive and
-    negative values::
-
-        -log(1 + exp(-x_i))     if x_i > 0
-        x_i - log(1 + exp(x_i)) if x_i <= 0
-
-    For the ordinary logistic function, use ``scipy.special.expit``.
-
-    Parameters
-    ----------
-    X : array-like of shape (M, N) or (M,)
-        Argument to the logistic function.
-
-    out : array-like of shape (M, N) or (M,), default=None
-        Preallocated output array.
-
-    Returns
-    -------
-    out : ndarray of shape (M, N) or (M,)
-        Log of the logistic function evaluated at every point in x.
-
-    Notes
-    -----
-    See the blog post describing this implementation:
-    http://fa.bianp.net/blog/2013/numerical-optimizers-for-logistic-regression/
-    """
-    is_1d = X.ndim == 1
```

Comment:
This function was still "kind" of public. It can internally call `np.logaddexp` now.
Can we deprecate it and asking to use the numpy one and then we remove it in 2 versions.

Reply:
But it is not listed in our API docs. So it is not public.
Do you know of any user/3rd party using that function from us?

Reply:
https://github.com/search?q=log_logistic+language%3APython&type=code&l=Python&p=1 gives 904 hits, most of which are forks of scikit-learn. For me, that's strong evidence that we can remove. I can add a changelog and say with what function to replace it.

Reply:
> But it is not listed in our API docs.

Yes, we have some legacy and bad practices that should be fixed. We should have clear public, developer, and private tools.

> So it is not public.

This is tricky because the public/private agreement would be the presence or not of a leading `_`.

> Do you know of any user/3rd party using that function from us?

Apparently some do: https://github.com/search?q=from+sklearn.utils.extmath+import+log_logistic&type=code

Reply:
What do you propose? Keep it with wrapping logaddexp and deprecate it? (This is what turns a dead simple PR in a maintenance nightmare. Just my very frank opinion.)

Reply:
About vowpalwabbit, their only use is `np.exp(log_logistic(...))` which is just crazy.

Reply:
> Keep it with wrapping logaddexp and deprecate it?

Yes.
```

Reply:
Not happy but will do.

Reply:
As you see, that's not a 5 minutes thing. It cost me 1/2h.

Reply:
But that prevented potential breakages which my hasty approving review definitely had not envisioned. :facepalm: 

Thank you for the efforts, @lorentzenchr.